bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -3631,6 +3631,7 @@ void RewriteInstance::mapCodeSections(BOLTLinker::SectionMapper MapSection) {
     Function.setImageAddress(FuncSection->getAllocAddress());
     Function.setImageSize(FuncSection->getOutputSize());
     if (Function.getImageSize() > Function.getMaxSize()) {
+      assert(!BC->isX86() && "Unexpected large function.");
```

Comment:
@yota9, @mtvec – would it be correct to check if the function has constant islands instead?

Reply:
To clarify the question, could you test it on your side with assertion changed to check for CIs? Ideally, the check should be properly modified to accommodate for CIs on supported architectures.